### ISSUE TITLE
Add util for random_string with special characters

### ISSUE THREAD

**Issue by saraburns1** [overhangio/tutor#1177]
https://github.com/overhangio/tutor/blob/0b934aff639fdd5c63ad859075c59ebea3b6a255/tutor/utils.py#L69

Add new util `random_string_special` that includes string.punctuation(https://docs.python.org/3/library/string.html#string.punctuation)

This is needed when using random_string to create Clickhouse passwords

**Comment by regisb** [comment]
I'm guessing this is required in the context of Aspects, right? Have you considered using the [ENV_TEMPLATE_VARIABLES](https://docs.tutor.edly.io/reference/api/hooks/catalog.html#tutor.hooks.Filters.ENV_TEMPLATE_VARIABLES) filter?

**Comment by saraburns1** [comment]
Yes for Aspects. I haven't considered that, but I'm not sure if it helps - I still need to generate random strings but with special characters included. This is what we're doing now ```("ASPECTS_CLICKHOUSE_LRS_PASSWORD", "{{ 24|random_string }}"),```

**Comment by regisb** [comment]
You could implement what you need with the following:

```
def random_string_special(length):
    return "".join([random.choice(string.ascii_letters + string.digits + string.punctuation) for _ in range(lenth)])

hooks.Filters.ENV_TEMPLATE_VARIABLES.add_item(("random_string_special", random_string_special))
```

And then, in configuration:

```
("ASPECTS_CLICKHOUSE_LRS_PASSWORD", "{{ 24|random_string_special }}"),
```

I'm not saying we should not include "random_string_special" in Tutor core. I just want to highlight that if you need it now in Aspects, then you can add to in your plugin.

FYI you should be careful about inclusion of such strings in templates. For instance, quote signs will not be escaped, which means that you should always wrap those strings in double quotes "...".

**Comment by saraburns1** [comment]
Oh, perfect! I was going to just write a new util function but this is cleaner. Thank you!

**Comment by DawoudSheraz** [comment]
@saraburns1 Hi, is this good to close, considering Régis has already added a path forward [above](https://github.com/overhangio/tutor/issues/1177#issuecomment-2538352152). Let me know if you have any questions. Thanks

**Comment by saraburns1** [comment]
Good to close - thanks!
